src/components/App.js

```diff
@@ -1,8 +1,196 @@
+import { wordDictionary2 } from '../letters/WordDictionary2';
+import { wordDictionary3 } from '../letters/WordDictionary3';
+import { wordDictionary4 } from '../letters/WordDictionary4';
+import { wordDictionary5 } from '../letters/WordDictionary5';
+import { letterPoints } from '../letters/LetterPoints';
 
 function App() {
+  const [selectNumber, setSelectNumber] = useState(false);
+  const [wordNo, setWordNo] = useState(1);
+  const [numberSelected, setNumberSelected] = useState('');
+  const [squares, setSquares] = useState([]);
+  const [wordLengths, setWordLengths] = useState([]);
+  const [remainingAlphabet, setRemainingAlphabet] = useState(alphabet);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [score, setScore] = useState(0);
+  let game;
+  let words = [];
+  let invalidWord = '';
+
+  class Game {
+    restart() {
+      setSelectNumber(false);
+      setNumberSelected('');
+      setWordLengths([]);
+      setSquares([]);
+      setWordNo(1);
+      setErrorMessage('');
+      setRemainingAlphabet(alphabet);
+      setScore(0);
+      console.clear();
+    }
+    calculateScore(words) {
+      let workScore = 20 - numberSelected;
+      for (let i = 0; i < words.length; i++) {
+        for (let j = 0; j < words[i].length; j++) {
+          workScore =
+            workScore +
+            letterPoints.find(item => {
+              return item.letter === words[i][j].toUpperCase();
+            }).point;
+      }
+    }
+    return workScore;
+    }
+  };
+  game = new Game();
+
+  const handleClick = () => {
+    // which squares contain 1's,2's,3's,.....
+    let wordN;
+    let result = '';
+    for (let i = 1; i < wordNo + 1; i++) {
+      wordN = [];
+      for (let j = 0; j < squares.length; j++) {
+        wordN = squares.filter(event => {
+          if (event.wordNums.length === 1) {
+            if (event.wordNums[0] === i) {
+              return true;
+            } else {
+              return false;
+            }
+          } else if (event.wordNums.length === 2) {
+            if (event.wordNums[0] === i || event.wordNums[1] === i) {
+              return true;
+            } else {
+              return false;
+            }
+          } else {
+            return false;
+          }
+        });
+      }
+      // combine words from board
+      result = '';
+      for (let k = 0; k < wordN.length; k++) {
+        result = result + wordN[k].letter;
+      }
+      words.push(result);
+    }
+    verifyBoard(words);
+  };
+
+  function checkWords() {
+    let allValid = true;
+    //let invalidWord = '';
+    for (let j = 0; j < words.length; j++) {
+      if (validWord(words[j]) === false) {
+        allValid = false;
+        invalidWord = words[j];
+        break;
+      }
+    }
+    return allValid
+  }
+
+  function verifyBoard(words) {
+    let workErrorMessage = '';
+    // Are all squares filled in?
+    for (let j = 0; j < squares.length; j++) {
```

Comment:
most of your `for loops` can be simplified to use js array functions. For example here you could use `.some`

Reply:
I converted 2 for loops to .some logic. I see more loops but I don't see how I could convert them to some.

Reply:
I converted some for loops into forEach.

Reply:
I think the ones that can be converted are.